prdoc/pr_9125.prdoc

```diff
@@ -0,0 +1,67 @@
+title: '[Xcm Precompile] [Solution 2] Revert xcm precompile call with a verbose error'
+doc:
+- audience: Runtime User
+  description: |-
+    This PR improves error handling for the XCM precompile by returning detailed error information instead of generic error messages.
+
+
+    before
+
+    ```rust
+    fn revert(error: &impl fmt::Debug, message: &str) -> Error {
+        error!(target: LOG_TARGET, ?error, "{}", message);
+        Error::Revert(message.into())
+    }
+    ```
+
+
+    after
+    ```rust
+
+    fn revert(error: &impl fmt::Debug, message: &str) -> Error {
+        error!(target: LOG_TARGET, ?error, "{}", message);
+        Error::Revert(Revert {
+            reason: format!("{:?}",error)
+        })
```

Comment:
is this `Error::Revert` visible on-chain? is it visible to the calling SC?

if visible to SC, we have to keep it as simple and compatible as possible because it should never change. A string or complex error-type (like dispatch errors) are NO-GO.

if it is only visible to debug tooling and we can freely change the error content at any time, then this is ok